File: app/src/main/java/com/manasmalla/ahamsvasth/AhamSvasthaUser.java
package com.manasmalla.ahamsvasth;

import android.content.Context;
import android.content.SharedPreferences;

import java.util.HashMap;
import java.util.function.Function;

public class AhamSvasthaUser {
    String gender = "";
    int age = 0;
    float height, weight;
    String unit, wUnit;
    HashMap<String, Boolean> diseases;
    String activityUser;

    public void setGender(String gender){
        this.gender = gender;
    }
    public void setAge(int age){
        this.age = age;
    }
    public void setHeight(float height){
        this.height = height;
    }
    public void setUnit(String unit){
        this.unit = unit;
    }
    public String getHeight(){
        return String.valueOf(this.height) + unit;
    }
    public void setWeight(float weight){
        this.weight = weight;
    }
    public void setWUnit(String wunit){
        this.wUnit = wunit;
    }
    public String getWeight(){
        return String.valueOf(this.weight) + wUnit;
    }
    public void setDiseases(HashMap<String, Boolean> diseases){
        this.diseases = diseases;
    }
    public String getDiseases(){
        return this.diseases.toString();
    }
    public HashMap<String, Boolean> getDiseasesHashMap(){
        return this.diseases;
    }
    public void setUserActivity(String activity){
        this.activityUser = activity;
    }
    public String getUserData(){
        return "Gender: " + gender + "; Age: " + age + "years; Height: " + getHeight() + "; Weight: " + getWeight() + "; Diseases: " + getDiseases() + "; User is " + activityUser + "!";
    }
    public static void saveAhamSvasthaUser(AhamSvasthaUser user, Context context, String username){

        SharedPreferences sharedPreferences = context.getSharedPreferences("com.manasmalla.ahamsvasth", Context.MODE_PRIVATE);
        sharedPreferences.edit().putString("Gender"+username, user.gender).commit();
        sharedPreferences.edit().putInt("Age"+username, user.age).commit();
        sharedPreferences.edit().putFloat("Height"+username, user.height).commit();
        sharedPreferences.edit().putString("hUnit"+username, user.unit).commit();
        sharedPreferences.edit().putFloat("Weight"+username, user.weight).commit();
        sharedPreferences.edit().putString("wUnit"+username, user.wUnit).commit();
        sharedPreferences.edit().putBoolean("isHavingDiabetes"+username, user.diseases.get("Diabetes")).commit();
        sharedPreferences.edit().putBoolean("isHavingThyroid"+username, user.diseases.get("Thyroid")).commit();
        sharedPreferences.edit().putBoolean("isHavingCholestrol"+username, user.diseases.get("Cholestrol")).commit();
        sharedPreferences.edit().putBoolean("isHavingBP"+username, user.diseases.get("BP")).commit();
        sharedPreferences.edit().putBoolean("isHavingObesity"+username, user.diseases.get("Obesity")).commit();
        sharedPreferences.edit().putString("userActivity"+username, user.activityUser).commit();
    }
    public static AhamSvasthaUser getAhamSvasthaUser(Context context, String username){
        AhamSvasthaUser restoredUser = new AhamSvasthaUser();
        SharedPreferences sharedPreferences = context.getSharedPreferences("com.manasmalla.ahamsvasth", Context.MODE_PRIVATE);
        restoredUser.setGender(sharedPreferences.getString("Gender"+username, null));
        restoredUser.setAge(sharedPreferences.getInt("Age"+username, 0));
        restoredUser.setHeight(sharedPreferences.getFloat("Height"+username, 0));
        restoredUser.setUnit(sharedPreferences.getString("hUnit"+username, null));
        restoredUser.setWeight(sharedPreferences.getFloat("Weight"+username,0));
        restoredUser.setWUnit(sharedPreferences.getString("wUnit"+username, null));
        HashMap<String, Boolean> restoredDiseases = new HashMap<>();
        if (sharedPreferences.getBoolean("isHavingDiabetes"+username, false)){
            restoredDiseases.put("Diabetes", true);
        }else{
            restoredDiseases.put("Diabetes", false);
        }
        if (sharedPreferences.getBoolean("isHavingThyroid"+username, false)){
            restoredDiseases.put("Thyroid", true);
        }else{
            restoredDiseases.put("Thyroid", false);
        }
        if (sharedPreferences.getBoolean("isHavingCholestrol"+username, false)){
            restoredDiseases.put("Cholestrol", true);
        }else{
            restoredDiseases.put("Cholestrol", false);
        }
        if (sharedPreferences.getBoolean("isHavingBP"+username, false)){
            restoredDiseases.put("BP", true);
        }else{
            restoredDiseases.put("BP", false);
        }
        if (sharedPreferences.getBoolean("isHavingObesity"+username, false)){
            restoredDiseases.put("Obesity", true);
        }else{
            restoredDiseases.put("Obesity", false);
        }
        restoredUser.setDiseases(restoredDiseases);
        restoredUser.setUserActivity(sharedPreferences.getString("userActivity"+username, null));
        return restoredUser;
    }

    public static String getCurrentUsername(Context context){
        SharedPreferences sharedPreferences = context.getSharedPreferences("com.manasmalla.ahamsvasth", Context.MODE_PRIVATE);
        return sharedPreferences.getString("current_username", sharedPreferences.getString("name", null));
    }
}

File: app/src/main/java/com/manasmalla/ahamsvastha/AhamSvasthaUser.kt
package com.manasmalla.ahamsvastha

import android.annotation.SuppressLint
import android.content.Context
import android.content.SharedPreferences
import java.util.*


class AhamSvasthaUser {
    lateinit var name: String
    lateinit var username: String
    lateinit var password: String
    var age: Int = 0
    lateinit var gender: String
    var height: Double = 0.0
    var weight: Double = 0.0
    lateinit var diseases: HashMap<String, Boolean>
    lateinit var hUnit: String
    lateinit var wUnit: String
    lateinit var activityUser: String

    @JvmName("setName1")
    fun setName(name: String) {
        this.name = name
    }

    @JvmName("setUsername1")
    fun setUsername(username: String) {
        this.username = username
    }

    @JvmName("setPassword1")
    fun setPassword(password: String) {
        this.password = password
    }

    @JvmName("setAge1")
    fun setAge(age: Int) {
        this.age = age
    }

    @JvmName("setGender1")
    fun setGender(gender: String) {
        this.gender = gender
    }

    @JvmName("setHeight1")
    fun setHeight(height: Double) {
        this.height = height
    }

    @JvmName("setWeight1")
    fun setWeight(height: Double) {
        this.height = height
    }

    @JvmName("setDiseases1")
    fun setDiseases(diseasesList: HashMap<String, Boolean>) {
        this.diseases = diseasesList
    }

    @JvmName("setHUnit1")
    fun setHUnit(hUnit: String) {
        this.hUnit = hUnit
    }

    @JvmName("setWUnit1")
    fun setWUnit(wUnit: String) {
        this.wUnit = wUnit
    }

    @JvmName("getHeight1")
    fun getHeight(): String {
        return "$height" + hUnit
    }

    @JvmName("getWeight1")
    fun getWeight(): String {
        return "$weight" + wUnit
    }

    fun getDiseases(): String {
        return this.diseases.toString();
    }

    fun getDiseasesHashMap(): HashMap<String, Boolean> {
        return diseases
    }

    fun setUserActivity(userActivity: String) {
        this.activityUser = userActivity
    }

    fun getUserData(): String {
        return "Gender: $gender ; Age: $age years; Height: ${getHeight()}; Weight: ${getWeight()}; Diseases:${getDiseases()}; User is $activityUser!"
    }

    @SuppressLint("ApplySharedPref")
    fun saveAhamSvasthaUser(user: AhamSvasthaUser, context: Context, username: String) {
        val sharedPreferences: SharedPreferences =
            context.getSharedPreferences("com.manasmalla.ahamsvastha", Context.MODE_PRIVATE)
        sharedPreferences.edit().putString("Gender$username", user.gender).commit()
        sharedPreferences.edit().putInt("Age$username", user.age).commit()
        sharedPreferences.edit().putFloat("Height$username", user.height.toFloat()).commit()
        sharedPreferences.edit().putString("hUnit$username", user.hUnit).commit()
        sharedPreferences.edit().putFloat("Weight$username", user.weight.toFloat()).commit()
        sharedPreferences.edit().putString("wUnit$username", user.wUnit).commit()
        sharedPreferences.edit()
            .putBoolean("isHavingDiabetes$username", user.diseases["Diabetes"]!!).commit()
        sharedPreferences.edit().putBoolean("isHavingThyroid$username", user.diseases["Thyroid"]!!)
            .commit()
        sharedPreferences.edit()
            .putBoolean("isHavingCholestrol$username", user.diseases["Cholestrol"]!!).commit()
        sharedPreferences.edit().putBoolean("isHavingBP$username", user.diseases["BP"]!!).commit()
        sharedPreferences.edit().putBoolean("isHavingObesity$username", user.diseases["Obesity"]!!)
            .commit()
        sharedPreferences.edit().putString("userActivity$username", user.activityUser).commit()
        sharedPreferences.edit().putString("currentUserUsername", username).commit()
        sharedPreferences.edit().putInt("numberOfUsers", sharedPreferences.getInt("numberOfUsers", 0) + 1).commit()
        if (sharedPreferences.getStringSet("listOfUsernames", null) != null){
            val listStringSet = sharedPreferences.getStringSet("listOfUsernames", null)
            listStringSet!!.add(username)
            sharedPreferences.edit().putStringSet("listOfUsernames", listStringSet).commit()
        }else{
            sharedPreferences.edit().putStringSet("listOfUsernames", setOf("$username")).commit()
        }
    }

    fun getAhamSvasthaUser(context: Context, username: String): AhamSvasthaUser? {
        val restoredUser = AhamSvasthaUser()
        val sharedPreferences: SharedPreferences =
            context.getSharedPreferences("com.manasmalla.ahamsvastha", Context.MODE_PRIVATE)
        restoredUser.setGender(sharedPreferences.getString("Gender$username", null)!!)
        restoredUser.setAge(sharedPreferences.getInt("Age$username", 0))
        restoredUser.setHeight(sharedPreferences.getFloat("Height$username", 0f).toDouble())
        restoredUser.setHUnit(sharedPreferences.getString("hUnit$username", "")!!)
        restoredUser.setWeight(sharedPreferences.getFloat("Weight$username", 0f).toDouble())
        restoredUser.setWUnit(sharedPreferences.getString("wUnit$username", "")!!)
        val restoredDiseases = HashMap<String, Boolean>()
        restoredDiseases["Diabetes"] =
            sharedPreferences.getBoolean("isHavingDiabetes$username", false)
        restoredDiseases["Thyroid"] =
            sharedPreferences.getBoolean("isHavingThyroid$username", false)
        restoredDiseases["Cholestrol"] =
            sharedPreferences.getBoolean("isHavingCholestrol$username", false)
        restoredDiseases["BP"] = sharedPreferences.getBoolean("isHavingBP$username", false)
        restoredDiseases["Obesity"] =
            sharedPreferences.getBoolean("isHavingObesity$username", false)
        restoredUser.setDiseases(restoredDiseases)
        restoredUser.setUserActivity(sharedPreferences.getString("userActivity$username", null)!!)
        return restoredUser
    }

    fun getAhamSvasthaUserNumber(ctx: Context):Int{
        var userList = 0
        val sharedPreferences: SharedPreferences = ctx.getSharedPreferences("com.manasmalla.ahamsvastha", Context.MODE_PRIVATE)
        userList = sharedPreferences.getInt("numberOfUsers", 0)
        return userList
    }

    fun getCurrentUser(ctx: Context):String{
        lateinit var userName:String
        val sharedPreferences: SharedPreferences = ctx.getSharedPreferences("com.manasmalla.ahamsvastha", Context.MODE_PRIVATE)
        userName = sharedPreferences.getString("currentUserUsername", "")!!
        return userName
    }

    fun getAhamSvasthaUserList(ctx: Context):Set<String>{
        val sharedPreferences: SharedPreferences = ctx.getSharedPreferences("com.manasmalla.ahamsvastha", Context.MODE_PRIVATE)
        return sharedPreferences.getStringSet("listOfUsernames", null)!!
    }

}